src/components/number-picker/number-picker.component.ts

```diff
@@ -49,10 +49,11 @@ export class NumberPickerComponent implements ControlValueAccessor, OnDestroy, O
     }
 
     set value(value: number) {
-        this._value = value;
-        this.valueChange.emit(value);
-        this._propagateChange(value);
-        this._valid = this.isValid();
+        if (this._value !== value) {
+            this._value = value;
+            this._propagateChange(value);
```

Comment:
I think this should also be in the increment and decrement functions as this is essentially the equivalent of an event emitter for Angular Forms and could mark a field as `dirty` whenever it hasn't actually been changed